### ISSUE TITLE
ActiveRecord::RecordNotFound on #show with load_and_authorize_resource

### ISSUE THREAD

**Issue by prussiap** [CanCanCommunity/cancancan#99]
Hey guys,
I've been playing around with cancancan/devise and rails 4 to build a wiki app. I've tested my abilities files and all the abilities i wanted seem to work fine there and in console.

I'm falling asleep so it's possible i'm just missing this or just an idiotic thing on my part. Any help is appreciated
(ps rspec for #index shows the correct pages and tests properly signed in)

Unfortunately I'm trying to test a very simple #show and am getting:

```
 1) PagesController#show finds owned page
     Failure/Error: get :show, id: page.id
     ActiveRecord::RecordNotFound:
       ActiveRecord::RecordNotFound
```

My index test worked fine and when I inspect my my page and user I get an id and they are properly built. 

I suspect after reading [this](https://github.com/ryanb/cancan/issues/43) that i has to do with find_by_id not working or somehow not loading the proper pages. 

I went explicit with the rspec:

```

    it "finds owned page" do
      user = create(:admin_user)
      page = create(:private_page, user_id: user.id)
      user2 = create(:editor_user)
      page2 = create(:private_page, user_id: user2.id)
      p page.inspect
      sign_in user
      get :show, id: page.id
      p "hellothere"
      assigns(:page).should == public_page_with_editor
    end
```

controller:

```
class PagesController < ApplicationController
  load_and_authorize_resource :find_by => :slug 

  def show
    @body = WikiCloth::Parser.new(:data => @page.body)
  end
end

```

Ability:

```
class Ability
  include CanCan::Ability

  def initialize(user)
    alias_action :read, :update, :to => :edit_page
    alias_action :read, :update, :create, :destroy, :to => :crud_page
    user ||= User.new # guest user (not logged in)

    case user.roles.first
    when :super_admin
      can :manage, :all
    when :admin
      can :manage, Page
    when :editor
      can :manage, Page, user_id: user.id
      can :edit_page, Page, page_rights: { user_id: user.id, right: 'read' }
      can :edit_page, Page, page_rights: { user_id: user.id, right: 'edit_page' }
      can :crud_page, Page, page_rights: { user_id: user.id, right: 'crud_page' }
      can :read, Page, public_flag: true
    else
      can :read, Page, public_flag: true
    end
  end

end
```

If you need user too let me know

```
class Page < ActiveRecord::Base
  extend FriendlyId
  belongs_to :user
  has_many :page_rights
  has_many :users, through: :page_rights

  friendly_id :title, use: :slugged
  validates_presence_of :title, :body, :slug
  acts_as_taggable
  has_paper_trail

  validates_inclusion_of :public_flag, :in => [true, false]
end
```

**Comment by coorasse** [comment]
Closed because of starvation. 
Please feel free to reopen if this issue is still valid.

**Comment by JohnPettigrew** [comment]
I'm seeing exactly the same problem as was reported here a few years ago: friendlyid works IFF cancancan isn't running load_and_authorize_resource in the controller. If I disable the call, friendlyid works. If I enable it, I get the RecordNotFound error again. Can anyone suggest a fix? (Devise 4.6.2, Cancancan 3.0.1, FriendlyId 5.2.5, Rails 5.2.3.)
